src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/PriceFilter.php

```diff
@@ -0,0 +1,250 @@
+     *
+     * @throws InvalidPropertyTypeException
+     * @throws InvalidPropertyException
+     */
+    protected function checkValue(AttributeInterface $attribute, $data)
+    {
+        if (!is_array($data)) {
+            throw InvalidPropertyTypeException::arrayExpected($attribute->getCode(), static::class, $data);
+        }
+
+        if (!array_key_exists('amount', $data)) {
+            throw InvalidPropertyTypeException::arrayKeyExpected(
+                $attribute->getCode(),
+                'amount',
+                static::class,
+                $data
+            );
+        }
+
+        if (!array_key_exists('currency', $data)) {
+            throw InvalidPropertyTypeException::arrayKeyExpected(
+                $attribute->getCode(),
+                'currency',
+                static::class,
+                $data
+            );
+        }
+
+        if (null !== $data['amount'] && !is_numeric($data['amount'])) {
+            throw InvalidPropertyTypeException::validArrayStructureExpected(
+                $attribute->getCode(),
+                sprintf('key "amount" has to be a numeric, "%s" given', gettype($data['amount'])),
+                static::class,
+                $data
+            );
+        }
+
+        $this->checkCurrency($attribute, $data);
+    }
+
+    /**
+     * @param AttributeInterface $attribute
+     * @param array              $data
+     *
+     * @throws InvalidPropertyTypeException
+     * @throws InvalidPropertyException
+     */
+    protected function checkCurrency(AttributeInterface $attribute, $data)
+    {
+        if (!array_key_exists('currency', $data) || !is_string($data['currency'])) {
+            throw InvalidPropertyTypeException::validArrayStructureExpected(
+                $attribute->getCode(),
+                sprintf('key "currency" has to be a string, "%s" given', gettype($data['currency'])),
+                static::class,
+                $data
+            );
+        }
+
+        if (!in_array($data['currency'], $this->currencyRepository->getActivatedCurrencyCodes()) &&
+            '' !== $data['currency']
+        ) {
+            throw InvalidPropertyException::validEntityCodeExpected(
+                $attribute->getCode(),
+                'currency',
+                'The currency does not exist',
+                static::class,
+                $data['currency']
+            );
+        }
+    }
+
+    /**
+     * Returns the attribute path for currency
+     *
+     * @param AttributeInterface $attribute
+     * @param string             $locale
+     * @param string             $channel
+     * @param array              $value
+     *
+     * @return string
+     */
+    protected function getAttributePathWithCurrency($attribute, $locale, $channel, array $value)
+    {
+        $attributePath = $this->getAttributePath($attribute, $locale, $channel);
+
+        if (null !== $value['currency'] && '' !== $value['currency']) {
```

Comment:
This is possible with the isEmpty operator for instance. The functional case would be: Give me products where the price does not exists for all currencies vice versa for Not Empty.
But I agree the naming is somewhat misleading